File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/cior/d/CiorapppathgyAggDO.java
package iih.ci.ord.cior.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.ord.cior.d.desc.CiorapppathgyAggDODesc;



/**
 * 病理申请单
 */
public class CiorapppathgyAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public OrdApPathgyDO getParentDO() {
		return ((OrdApPathgyDO) super.getParentDO());
	}

	public void setParentDO(OrdApPathgyDO headDO) {
		setParent(headDO);
	}

	public OrdApPathgySampDO[] getOrdApPathgySampDO() {
		IBaseDO[] dos = getChildren(OrdApPathgySampDO.class);
		if(dos==null) return null;
		OrdApPathgySampDO[] result=new OrdApPathgySampDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(OrdApPathgySampDO)dos[i];
		}
		return result;
	}
	
	public void setOrdApPathgySampDO(OrdApPathgySampDO[] dos) {
		setChildren(OrdApPathgySampDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		CiorapppathgyAggDODesc desc = new CiorapppathgyAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new OrdApPathgyDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.ord.cior.d.OrdApPathgySampDO")) {
             return new OrdApPathgySampDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.ord/cior/d/CiorapppathgyAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.ord.cior.d
{
	/// <summary>
	/// 病理申请单DO
	/// </summary>
	public class CiorapppathgyAggDO : BaseAggDO
	{	
	    private static string ORDAPPATHGYDO_ORDAPPATHGYSAMPDO="iih.ci.ord.cior.d.OrdApPathgySampDO";

        public CiorapppathgyAggDO() {
            this.setParentDO(new OrdApPathgyDO());
        }

  	    public OrdApPathgyDO getParentDO() {
		    return ((OrdApPathgyDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(OrdApPathgyDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public OrdApPathgySampDO[] getOrdApPathgySampDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(ORDAPPATHGYDO_ORDAPPATHGYSAMPDO));
		    if (dos == null || dos.Length==0){return null;}
            OrdApPathgySampDO[] r = new OrdApPathgySampDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as OrdApPathgySampDO;
            }
	        return r;
		    
	    }
	    
	    public void setOrdApPathgySampDO(OrdApPathgySampDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new OrdApPathgyDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (ORDAPPATHGYDO_ORDAPPATHGYSAMPDO == clzName)
	        {
                return new OrdApPathgySampDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.ord.cior.d.OrdApPathgySampDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
